operator/pkg/upgrade/upgrade.go

```diff
@@ -202,6 +202,31 @@ func updateClusterConfig(ctx context.Context, cli client.Client, in *ecv1beta1.I
 		}
 	}
 
+	// Apply unsupported overrides from the installation
+	if (in.Spec.Config != nil && in.Spec.Config.UnsupportedOverrides.K0s != "") || in.Spec.EndUserK0sConfigOverrides != "" {
```

Comment:
don't see why these two conditions aren't separate if blocks. but i won't block the pr for that.